packages/node/src/integrations/anr/index.ts

```diff
@@ -0,0 +1,155 @@
+// TODO (v8): This import can be removed once we only support Node with global URL
+import { URL } from 'url';
+import { getCurrentScope } from '@sentry/core';
+import type { Contexts, Event, EventHint, Integration } from '@sentry/types';
+import { dynamicRequire, logger } from '@sentry/utils';
+import type { Worker, WorkerOptions } from 'worker_threads';
+import type { NodeClient } from '../../client';
+import { NODE_VERSION } from '../../nodeVersion';
+import type { Options, WorkerStartData } from './common';
+import { base64WorkerScript } from './worker-script';
+
+const DEFAULT_INTERVAL = 50;
+const DEFAULT_HANG_THRESHOLD = 5000;
+
+type WorkerNodeV14 = Worker & { new (filename: string | URL, options?: WorkerOptions): Worker };
+
+type WorkerThreads = {
+  Worker: WorkerNodeV14;
+};
+
+function log(message: string, ...args: unknown[]): void {
+  logger.log(`[ANR] ${message}`, ...args);
+}
+
+/**
+ * We need to use dynamicRequire because worker_threads is not available in node < v12 and webpack error will when
+ * targeting those versions
+ */
+function getWorkerThreads(): WorkerThreads {
+  return dynamicRequire(module, 'worker_threads');
+}
+
+/**
+ * Gets contexts by calling all event processors. This relies on being called after all integrations are setup
+ */
+async function getContexts(client: NodeClient): Promise<Contexts> {
+  let event: Event | null = { message: 'ANR' };
+  const eventHint: EventHint = {};
+
+  for (const processor of client.getEventProcessors()) {
```

Comment:
should we also include global event processors here?

Reply:
Possibly 🤔

This is only really used to get the os/device context via the `Context` and `Modules` integrations.

Reply:
alright I think it's fine for now then, we're getting rid of global event processors anyway.